label.py

```diff
@@ -11,80 +11,106 @@
 
 
 def get_new_image():
-    global current_image
     if len(images_to_show) == 0:
-        root.quit()
+        cv2.destroyAllWindows()
         return
     current_image = images_to_show.pop()
-    print("data/%s_firefox.png" % current_image)
-
-    img_firefox = Image.open("data/%s_firefox.png" % current_image)
-    img_chrome = Image.open("data/%s_chrome.png" % current_image)
-
-    img = ImageTk.PhotoImage(img_firefox)
-    panel1.configure(image=img)
-    panel1.image = img
-    img = ImageTk.PhotoImage(img_chrome)
-    panel2.configure(image=img)
-    panel2.image = img
-
-    background = img_firefox.convert("RGBA")
-    overlay = img_chrome.convert("RGBA")
-    img_blended = Image.blend(background, overlay, 0.5)
-
-    img = ImageTk.PhotoImage(img_blended)
-    panel3.configure(image=img)
-    panel3.image = img
-
-
-# The images are the same.
-def callback_y(e):
-    labels[current_image] = 'y'
-    get_new_image()
-
-
-# The image are basically the same, except for advertisement or content.
-def callback_d(e):
-    labels[current_image] = 'd'
+    print("Reading %s" % current_image)
+    firefox_ss = cv2.imread("data/%s_firefox.png" % current_image)
+    chrome_ss = cv2.imread("data/%s_chrome.png" % current_image)
+    drawing_area_firefox = np.zeros((firefox_ss.shape), np.uint8)
+    drawing_area_chrome = np.zeros((chrome_ss.shape), np.uint8)
+    boxes_firefox = []
+    boxes_chrome = []
+    cv2.setMouseCallback('firefox', draw_boundary_box, [drawing_area_firefox, boxes_firefox])
+    cv2.setMouseCallback('chrome', draw_boundary_box, [drawing_area_chrome, boxes_chrome])
+    visiblity = 1
+    while True:
+        firefox_window = cv2.addWeighted(drawing_area_firefox, 1 - visiblity, firefox_ss, visiblity, 0)
+        chrome_window = cv2.addWeighted(drawing_area_chrome, 1 - visiblity, chrome_ss, visiblity, 0)
+        cv2.imshow('firefox', firefox_window)
+        cv2.imshow('chrome', chrome_window)
+        cv2.moveWindow('firefox', 20, 0)
+        cv2.moveWindow('chrome', 20 + firefox_window.shape[1], 0)
+        k = cv2.waitKey(1) & 0xFF
+        # <Escape> quits marking area without saving
+        if k == key_map["Escape"]:
+            cv2.destroyAllWindows()
+            return
+        # 'r' resets the present selection of boundary boxes
+        elif k == key_map["r"]:
+            drawing_area_firefox = np.zeros((firefox_ss.shape), np.uint8)
+            drawing_area_chrome = np.zeros((chrome_ss.shape), np.uint8)
+            boxes_chrome = []
+            boxes_firefox = []
+            cv2.setMouseCallback('firefox', draw_boundary_box, [drawing_area_firefox, boxes_firefox])
+            cv2.setMouseCallback('chrome', draw_boundary_box, [drawing_area_chrome, boxes_chrome])
+        # <Return> saves the current marking and moves to next image or skips it if no label is assigned
+        elif k == key_map["Enter"]:
+            if current_image not in labels.keys():
```

Comment:
When can this happen?

Reply:
Its for the same feature you added in `tkinter` based interface where if a person is unsure of the type and wants to skip marking that image .

Reply:
Maybe we should have a different key for skipping, to avoid confusion (Enter -> Finish bounding box marking, Space -> Skip image).

Reply:
Done!